lib/Test2/Harness/Runner.pm

```diff
@@ -413,7 +414,7 @@ sub next {
 
         if (my $lock = $self->lock_dispatch) {
             while (1) {
```

Comment:
This change essentially makes the nested while loop unnecessary. removing the `while` and the `last` would essentially be the same change. This makes me unsure if this fix might introduce a bug. I have to re-read the code and see if I can figure out why that second loop is there.